#161229849: Fetch security users emails from the api

Kalela opened the pull request:
#### What does this PR do?
Sets up correct endpoint for getting security users from the backend api.
#### Description of Task to be completed?
- Change base api to the new ` http://api-staging-art.andela.com ` from ` http://andela-learning.com `.
- Make getting the data from api responsive.
#### How should this be manually tested?
- Ensure you have a security account set up with the backend team.
- Open the app.
- Check the verbose logcat on Android studio and find the call made to security emails endpoint. You should find your email there.
- Log in with your security account.
#### What are the relevant pivotal tracker stories?
[#161229849](https://www.pivotaltracker.com/story/show/161229849)
A security user is fetched from the API
#### Questions:
None so far

Kalela committed:
feature(security users): Get security user emails from the api

- change base url to the backend format
- check if emails come back from api before updating the list

[Delivers: 161229849]

Kalela committed:
feature(security users): Get security user emails from the api

- change base url to the backend format
- check if emails come back from api before updating the list

[Delivers: 161229849]

doc2dev reviewed (changes requested):
The code in `LoginActivity.java` assumes that the security user emails would have been fetched by the time the Google sign-in is complete. This might not always be the case. Could you add checks to make sure that the API call has completed successfully by the time `isAllowedNonAndelaEmail` is called?

Kalela commented:
Working on it.

Kalela committed:
feature(security users): Make security users api call run before authentication happens:

- run security users api call to completion before running authentication
- remove merge errors

[Completes: 161229849]"

Kalela committed:
feature(security users): Run checks for emails recieved

- check if security user emails have been recieved before checking if the current email is in the list

[finishes: 161229849]
